### DISCUSSION TITLE
Git diff filter gives reverse output for D(deleted) and A (added)

### DISCUSSION THREAD

**Discussion by manoj798** [community/community#25997] [Actions]
<p>Hi,</p>
<p>I deleted two files (test1.json and test2.json) from my repository branch and run following command<br>
git diff --name-status current_commit_id previous_commit_id</p>
<p>I was expecting output<br>
D       test1.json<br>
D       test2.json</p>
<p>but I got actual out as<br>
A       test1.json<br>
A       test2.json</p>
<p>I ttried adding them back and then I got output</p>
<p>D       test1.json<br>
D       test2.json</p>
<p>Not sure if I am doing anything wrong… Any help?</p>

**Comment by manoj798** [comment]
<p>Hi,</p>
<p>I deleted two files (test1.json and test2.json) from my repository branch and run following command<br>
git diff --name-status current_commit_id previous_commit_id</p>
<p>I was expecting output<br>
D test1.json<br>
D test2.json</p>
<p>but I got actual out as<br>
A test1.json<br>
A test2.json</p>
<p>I ttried adding them back and then I got output</p>
<p>D test1.json<br>
D test2.json</p>
<p>Not sure if I am doing anything wrong… Any help?</p>
<p><em>[Staff edit: Merged duplicate topics]</em></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Order matters when comparing things with <code>git diff</code>. With <code>current_commit_id previous_commit_id</code> in that order you’ll see what changes there are going <em>from</em> <code>current_commit_id</code> <em>to</em> <code>previous_commit_id</code>, which would be your file getting added back in. Try this: :wink:</p>
<pre><code class="lang-bash">git diff --name-status previous_commit_id current_commit_id
</code></pre>

**Comment by moha1383** [comment]
<p><img src="https://user-images.githubusercontent.com/17278760/181094463-98596b26-38c9-4bf0-aa06-7f2d8dd706a1.gif" alt="894475738" data-base62-sha1="v1vipTTKxmrfj5QWL09wXQprUxc" width="195" height="185"></p>
